src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -490,12 +490,13 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
             CommunicationsUtilities.Trace("Launching node from {0}", msbuildLocation);
 
 #if RUNTIME_TYPE_NETCORE
-            string coreRunName = NativeMethodsShared.IsUnixLike ? "corerun" : "CoreRun.exe";
-            string exeName = Path.Combine(Path.GetDirectoryName(msbuildLocation), coreRunName);
+            // Try to get the process name in case of different host running.
```

Comment:
Maybe say "Run the child process with the same host as the currently-running process" to better capture intent?